File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3/QuoteContractOption3.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace QuoteContractOption3s

Public Class QuoteContractOption3

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "QuoteContractOption3 Properties"

      
Private _QuoteContractID As Integer = 0
Public Readonly Property QuoteContractID() As Integer
	Get 
		Return _QuoteContractID
	End Get	
End Property	
Public Writeonly Property SetQuoteContractID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteContractID", Value)
	End Set
End Property


Private _CustomerID As Integer = 0
Public Readonly Property CustomerID() As Integer
	Get 
		Return _CustomerID
	End Get	
End Property	
Public Writeonly Property SetCustomerID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_CustomerID", Value)
	End Set
End Property


Private _QuoteContractReference As String = String.Empty
Public Readonly Property QuoteContractReference() As String
	Get 
		Return _QuoteContractReference
	End Get	
End Property	
Public Writeonly Property SetQuoteContractReference() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteContractReference", Value)
	End Set
End Property


Private _QuoteContractStatusID As Integer = 0
Public Readonly Property QuoteContractStatusID() As Integer
	Get 
		Return _QuoteContractStatusID
	End Get	
End Property	
Public Writeonly Property SetQuoteContractStatusID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteContractStatusID", Value)
	End Set
End Property


Private _QuoteContractDate As Datetime = Nothing
Public Property QuoteContractDate() As Datetime
	Get 
		Return _QuoteContractDate
	End Get
	Set (ByVal Value As Datetime)
		_QuoteContractDate = Value
	End Set
End Property	


Private _QuoteContractPrice As Double = 0
Public Readonly Property QuoteContractPrice() As Double
	Get 
		Return _QuoteContractPrice
	End Get	
End Property	
Public Writeonly Property SetQuoteContractPrice() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteContractPrice", Value)
	End Set
End Property


Private _ReasonForReject As String = String.Empty
Public Readonly Property ReasonForReject() As String
	Get 
		Return _ReasonForReject
	End Get	
End Property	
Public Writeonly Property SetReasonForReject() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_ReasonForReject", Value)
	End Set
End Property

            Private _ReasonForRejectID As Integer = 0
            Public ReadOnly Property ReasonForRejectID() As Integer
                Get
                    Return _ReasonForRejectID
                End Get
            End Property
            Public WriteOnly Property SetReasonForRejectID() As Object
                Set(Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ReasonForRejectID", Value)
                End Set
            End Property
  
#End Region

End Class

End Namespace

End Namespace

File: FSM/Entity/QuoteContractOption3s/QuoteContractOption3.cs
// Decompiled with JetBrains decompiler
// Type: FSM.Entity.QuoteContractOption3s.QuoteContractOption3
// Assembly: FSM, Version=1.12.1.0, Culture=neutral, PublicKeyToken=null
// MVID: FAB6F0E2-B68A-47AB-AF95-A9E31DED7575
// Assembly location: C:\github\gabriel.fieldservicemanager\src\bin\x86\Debug\FSM.exe

using System;
using System.Collections;
using System.Runtime.CompilerServices;

namespace FSM.Entity.QuoteContractOption3s
{
    public class QuoteContractOption3
    {
        private DataTypeValidator _dataTypeValidator;
        private bool _exists;
        private bool _deleted;
        private int _QuoteContractID;
        private int _CustomerID;
        private string _QuoteContractReference;
        private int _QuoteContractStatusID;
        private DateTime _QuoteContractDate;
        private double _QuoteContractPrice;
        private string _ReasonForReject;
        private int _ReasonForRejectID;

        public QuoteContractOption3()
        {
            this._exists = false;
            this._deleted = false;
            this._QuoteContractID = 0;
            this._CustomerID = 0;
            this._QuoteContractReference = string.Empty;
            this._QuoteContractStatusID = 0;
            this._QuoteContractDate = DateTime.MinValue;
            this._QuoteContractPrice = 0.0;
            this._ReasonForReject = string.Empty;
            this._ReasonForRejectID = 0;
            this._dataTypeValidator = new DataTypeValidator();
        }

        public bool IgnoreExceptionsOnSetMethods
        {
            get
            {
                return this._dataTypeValidator.IgnoreExceptionsOnSetMethods;
            }
            set
            {
                this._dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
            }
        }

        public Hashtable Errors
        {
            get
            {
                return this._dataTypeValidator.Errors;
            }
        }

        public DataTypeValidator DTValidator
        {
            get
            {
                return this._dataTypeValidator;
            }
        }

        public bool Exists
        {
            get
            {
                return this._exists;
            }
            set
            {
                this._exists = value;
            }
        }

        public bool Deleted
        {
            get
            {
                return this._deleted;
            }
        }

        public bool SetDeleted
        {
            set
            {
                this._deleted = value;
            }
        }

        public int QuoteContractID
        {
            get
            {
                return this._QuoteContractID;
            }
        }

        public object SetQuoteContractID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_QuoteContractID", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public int CustomerID
        {
            get
            {
                return this._CustomerID;
            }
        }

        public object SetCustomerID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_CustomerID", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string QuoteContractReference
        {
            get
            {
                return this._QuoteContractReference;
            }
        }

        public object SetQuoteContractReference
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_QuoteContractReference", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public int QuoteContractStatusID
        {
            get
            {
                return this._QuoteContractStatusID;
            }
        }

        public object SetQuoteContractStatusID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_QuoteContractStatusID", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public DateTime QuoteContractDate
        {
            get
            {
                return this._QuoteContractDate;
            }
            set
            {
                this._QuoteContractDate = value;
            }
        }

        public double QuoteContractPrice
        {
            get
            {
                return this._QuoteContractPrice;
            }
        }

        public object SetQuoteContractPrice
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_QuoteContractPrice", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string ReasonForReject
        {
            get
            {
                return this._ReasonForReject;
            }
        }

        public object SetReasonForReject
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_ReasonForReject", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public int ReasonForRejectID
        {
            get
            {
                return this._ReasonForRejectID;
            }
        }

        public object SetReasonForRejectID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_ReasonForRejectID", RuntimeHelpers.GetObjectValue(value));
            }
        }
    }
}